src/Spago/Config.purs

```diff
@@ -334,18 +439,26 @@ readWorkspace { maybeSelectedPackage, pureBuild, migrateConfig } = do
             logDebug "Lockfile is up to date, using it"
             pure (Right contents)
 
-  -- Read in the package database
+loadPackageSet
+  :: ∀ a
+   . { workspaceConfig :: Core.WorkspaceConfig
+     , loadedPackages :: Map PackageName WorkspacePackage
+     , rootPath :: RootPath
+     , lockfile :: Either String Lockfile
+     }
+  -> Spago (Registry.RegistryEnv a) { packageSet :: PackageSet, compiler :: Range }
+loadPackageSet { lockfile, workspaceConfig, loadedPackages, rootPath } = do
   { offline } <- ask
-  packageSetInfo <- case maybeLockfileContents, workspace.packageSet of
+  packageSetInfo <- case lockfile, workspaceConfig.packageSet of
```

Comment:
(we tend to use [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation) for ease of brain-typechecking)
```suggestion
  packageSetInfo <- case eitherLockfile, workspaceConfig.packageSet of
```